freqtrade/strategy/interface.py

```diff
@@ -101,10 +101,10 @@ class IStrategy(ABC, HyperStrategyMixin):
     # run "populate_indicators" only for new candle
     process_only_new_candles: bool = False
 
-    use_sell_signal: bool
-    sell_profit_only: bool
-    sell_profit_offset: float
-    ignore_roi_if_buy_signal: bool
+    use_exit_signal: bool
+    exit_profit_only: bool
+    exit_profit_offset: float
+    ignore_roi_if_enter_signal: bool
```

Comment:
I am quite concerned about settings that are part of the strategy.

While you mark them as "deprecated" when coming from the configuration - settings like `use_sell_signal = True`  in the strategy are simply ignored.

That's the most dangerous of all approaches - as users will have a working strategy - and an update will silently ignore certain values - which may be required for the strategy to work.